napari/conftest.py

```diff
@@ -470,3 +472,166 @@ def _mock_app():
             yield app
         finally:
             Application.destroy('test_app')
+
+
+def _get_calling_place(depth=1):
+    if hasattr(sys, "_getframe"):
+        frame = sys._getframe(1 + depth)
+        return f"{frame.f_code.co_filename}:{frame.f_code.co_firstlineno}"
+    return ""
+
+
+@pytest.fixture
+def dangling_threads(monkeypatch):
+    from qtpy.QtCore import QThread
+
+    base_start = QThread.start
+    thread_dkt = WeakKeyDictionary()
+
+    def my_start(self, priority=QThread.InheritPriority):
+        thread_dkt[self] = _get_calling_place()
+        base_start(self, priority)
+
+    monkeypatch.setattr(QThread, 'start', my_start)
+    yield
+
+    dangling_threads = []
+
+    for thread, calling in thread_dkt.items():
+        try:
+            if thread.isRunning():
+                dangling_threads.append((thread, calling))
+        except RuntimeError as e:
+            if "wrapped C/C++ object of type QThread" not in e.args[0]:
+                raise
+
+    for thread, _ in dangling_threads:
+        thread.quit()
+        thread.wait(2000)
+
+    assert not dangling_threads, "thread called in:\n" + "\n".join(
+        x[1] for x in dangling_threads
+    )
+
+
+@pytest.fixture
+def dangling_thread_pool(monkeypatch):
+    from qtpy.QtCore import QThreadPool
+
+    base_start = QThreadPool.start
+    threadpool_dkt = WeakKeyDictionary()
+
+    def my_start(self, runnable, priority=0):
+        if self not in threadpool_dkt:
+            threadpool_dkt[self] = []
+        threadpool_dkt[self].append(_get_calling_place())
+        base_start(self, runnable, priority)
+
+    monkeypatch.setattr(QThreadPool, 'start', my_start)
+    yield
+
+    dangling_threads_pools = []
```

Comment:
Based on the [docs of `QThreadPool`](https://doc.qt.io/qt-5/qthreadpool.html#details), there should only be one pool per qt app. Should we expect more than one app to be running at once during setup/teardown of a test?

Reply:
No. The docs mean that there will be one global `QThreadPoll`. But You still could create the next one.

Reply:
Ah, true. This seems fine then.